fix default priority and clamp it

Gummibeer opened the pull request:
The default value of `priority` is `0.5` and it should be between 0 and 1. There is also no reason to use `0.8` as default cause the priority is just handled as relative value between all entries. 
https://www.sitemaps.org/de/protocol.html#prioritydef

brendt commented:
The snapshots test are broken in this PR. Take a look at the build: https://travis-ci.org/spatie/laravel-sitemap/jobs/355011067

If you're changing the priority, the snapshots should also be updated.

Gummibeer commented:
@brendt hey, sorry - just changed the files in browser and haven't run phpunit.
Have regenerated the snapshots and now it's passing all tests.

brendt commented:
For reference, the english version: https://www.sitemaps.org/protocol.html#prioritydef

brendt commented on src/Tags/Url.php:
```diff
@@ -87,7 +87,7 @@ public function setChangeFrequency(string $changeFrequency)
     public function setPriority(float $priority)
     {
-        $this->priority = $priority;
+        $this->priority = max(0, min(1, $priority));
```
I think it would be good to add a test for this newly added logic.

Gummibeer replied:
done https://github.com/spatie/laravel-sitemap/pull/144/commits/8450b5f9a4e09311cc2de15eddf7e23701ec386b

Gummibeer replied:
And to bring a usecase: if you generate a sitemap from your menu tree and say that every level reduces the priority by `0.2` and don't want to think about if it's valid.
* Home `1.0`
  * Cat1 `0.8`
    * Cat2 `0.6`
      * Cat3 `0.4`
        * Cat4 `0.2`
          * Cat5 `0.0`
            * Articles `-0.2` clamped to `0.0`

Gummibeer replied:
Or should I split it into two tests `priority_is_clamped_min` and `priority_is_clamped_max`?

brendt replied:
@Gummibeer this is fine!

brendt commented:
@Gummibeer Thanks, the PR seems good, but I'd like to see one more test, I've added a comment.

Gummibeer commented:
@brendt in https://github.com/spatie/laravel-sitemap/pull/144/commits/deb03b479b3cb844e71cf3c2b2a6a28f608e71bf I've added two `@test` annotations that were missing. Haven't checked all files, but these were in the same place where I've added the test https://github.com/spatie/laravel-sitemap/pull/144/commits/8450b5f9a4e09311cc2de15eddf7e23701ec386b.

brendt commented:
@Gummibeer Thanks! Unfortunately, we'll have to tag a major version for this change because some people might rely on the 0.8 default behaviour. 

We'll close the PR for now, but when we're ready for a new major version, we'll include it there.
